dags/sentinel-2-backlog-indexing.py

```diff
@@ -0,0 +1,82 @@
+"""
+# Sentinel-2 backlog indexing automation
+
+DAG to index Sentinel-2 backlog data.
+
+"""
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.kubernetes.secret import Secret
+from airflow.kubernetes.volume import Volume
+from airflow.kubernetes.volume_mount import VolumeMount
+from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+from airflow.operators.dummy_operator import DummyOperator
+
+from textwrap import dedent
+
+import kubernetes.client.models as k8s
+
+DEFAULT_ARGS = {
+    "owner": "ToktamEbadi",
+    "depends_on_past": False,
+    "start_date": datetime(2020, 6, 14),
+    "email": ["toktam.ebadi@ga.gov.au"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=5),
+    "env_vars": {
+        # TODO: Pass these via templated params in DAG Run
+        "DB_HOSTNAME": "db-writer",
+        "WMS_CONFIG_PATH": "/env/config/ows_cfg.py",
+        "DATACUBE_OWS_CFG": "config.ows_cfg.ows_cfg",
+    },
+    # Lift secrets into environment variables
+    "secrets": [
+        Secret("env", "DB_USERNAME", "odc-writer", "postgres-username"),
+        Secret("env", "DB_PASSWORD", "odc-writer", "postgres-password"),
+        Secret("env", "AWS_DEFAULT_REGION", "sentinel-2-indexing-user", "AWS_DEFAULT_REGION"),
+        Secret("env", "AWS_ACCESS_KEY_ID", "sentinel-2-indexing-user", "AWS_ACCESS_KEY_ID"),
+        Secret("env", "AWS_SECRET_ACCESS_KEY", "sentinel-2-indexing-user", "AWS_SECRET_ACCESS_KEY"),
+        Secret("env", "DB_DATABASE", "odc-writer", "database-name"),
+    ],
+}
+
+INDEXER_IMAGE = "opendatacube/datacube-index:latest"
+
+dag = DAG(
+    "Sentinel-2_indexing",
+    doc_md=__doc__,
+    default_args=DEFAULT_ARGS,
+    schedule_interval="@once",
+    catchup=False,
+    tags=["k8s", "Sentinel-2-indexing"],
+)
+
+with DAG('Sentinel-2-backlog-indexing', default_args=DEFAULT_ARGS,
+         schedule_interval=DEFAULT_ARGS['schedule_interval'],
+         tags=["Sentinel-2", "indexing"], catchup=False) as dag:
+
+
+    # This needs to be updated in the future in case more zones have been added
+    utm_zones = range(26, 42)
```

Comment:
how big is this, is it really better to startup 16 pods? i mean if each pod runs for 1hour that's real saver, but if it only running for 5minutes, then maybe not.

Reply:
Some of theses will run for half a day and some other ones for an hour.  It depends on the number of scenes in each zone.

I though this does that, I've looked at Grafana and (even Airflow), each of these zones will run on a separate work node/pod.  Am I missing anything?

Reply:
it does, it will dynamically create the pods for each of the UTM zones,